### PR TITLE
mysql8 layout fix

### PR DESCRIPTION
**Pull request by BjarneDMat**
#### Description

    On branch mysql8-layout-fix
    Changes to be committed:
            modified:   databases/mysql8/Portfile
    
    WorkAround for
            /opt/local/lib/mysql8/bin/my_openssl
    linking to openssl lib in
            /opt/local/lib/mysql8/bin/../lib/
    
    It's an upstream problem, in that my_openssl
    1) copies the openssl libs to /opt/local/lib/mysql8/mysql
    2) links to these libs in /opt/local/lib/mysql8lib/
    3) doesn't link to the openssl libs where these are installed

    On branch mysql8-layout-fix
    Changes to be committed:
            modified:   aqua/qt6/Portfile
            modified:   databases/mysql8/files/patch-cmake-install_layout.cmake.diff
            modified:   gis/gdal/Portfile
    
    reverses :
        [eeeb795e625ad50f416e8f69ff9dc520f7021c80/macports-ports] mysql8
        [dd5b391e5a3c4263f3457732b5f7c09ceec089b9/macports-ports] (qt6)
        [a7a4334c2169af6db0a69c8b1627bcc196acd69d/macports-ports] (gdal)

closes https://trac.macports.org/ticket/70001

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: databases/mysql8/Portfile [hunk 1]
```diff
@@ -209,6 +209,7 @@ if {$subport eq $name} {
             copy ${prefix}/share/${name_mysql}/support-files/macports/my.cnf \
                 ${prefix}/etc/${name_mysql}/
         }
+        file link -symbolic ${prefix}/lib/${name_mysql}/lib mysql
```
Comment (ryandesign) [inline databases/mysql8/Portfile:212]: This changes what the port installs so its revision needs to be increased. However it seems like it should be possible to fix this problem without creating a symlink.

**Review by ryandesign — COMMENTED** [review]
> followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?

The commit messages do not follow the guidelines.

> It's an upstream problem, in that my_openssl
> 1) copies the openssl libs to /opt/local/lib/mysql8/mysql
> 2) links to these libs in /opt/local/lib/mysql8lib/
> 3) doesn't link to the openssl libs where these are installed

Do we know why it does this? Can we stop it from doing this?

File: databases/mysql8/Portfile (hunk 1 above)
Comment (BjarneDMat) [reply databases/mysql8/Portfile:212]: The proposed fix is a WorkAround. The basic problem seems to be in my_openssl. I don't know enough about C/C++ programming to be able to solve it.

Also, I'm not that well versed in the macports process, so please forgive any errors I make

**Comment by jhoyt4** [comment]
> Do we know why it does this? Can we stop it from doing this?

@ryandesign - I have added commentary to the [track ticket](https://trac.macports.org/ticket/70001) to help define the issue better.  The short answer is mysql8's cmake files got a lot of homebrew based logic / hard codes added.

I have attempted to resolve these in a new [PR 24615](https://github.com/macports/macports-ports/pull/24615)

**Comment by jhoyt4** [comment]
@BjarneDMat - [PR24615](https://github.com/macports/macports-ports/pull/24615) was accepted, as that PR directly fixes the issue, I believe it makes sense to close this one.  Thank you for pointing identifying some of the issues that needed to be addresses.

**Comment by BjarneDMat** [comment]
replaced w/ https://github.com/macports/macports-ports/pull/24615

**Pull request closed without merging**